client/src/components/AuthForm.js

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useRef } from "react";
+import PropTypes from "prop-types";
+import TextField from "@material-ui/core/TextField";
+import { makeStyles } from "@material-ui/core/styles";
+import { Button } from "@material-ui/core";
+import { useAuth } from "../utils/AuthProvider";
+import _ from "lodash";
+const useStyles = makeStyles(theme => ({
+    root: {},
+    container: {},
+    textField: {},
+
+    displayName: {}
+}));
+
+const AuthForm = props => {
+    const { name, displayName } = props;
+    const classes = useStyles();
+    const [authForm, setAuthForm] = useState({
+        email: "",
+        password: "",
+        name: "",
+        confirmPassword: ""
+    });
+
+    const pwdRef = useRef(null);
+    const pwdRepeatRef = useRef(null);
+    const handleChange = elName => event => {
+        if (name === "signup") {
+            if (
+                (elName === "password" || elName === "confirmPassword") &&
+                event.target.value &&
+                event.target.value.length < 6
+            ) {
+                event.target.setCustomValidity(
+                    "Passwords have to be atleast 6 characters long"
+                );
```

Comment:
let's not modify the DOM here, and instead have a state variable for errors, and show the errors 
This is an anti-pattern in React. We want to have React do DOM updates as opposed to us doing them directly

Reply:
good point, I've updated the code to now use material ui's helper text instead